src/rules/no-extraneous-dependencies.js

```diff
@@ -73,19 +82,18 @@ function getDependencies(context, packageDir) {
       )
     }
 
-    if (![
-      packageContent.dependencies,
-      packageContent.devDependencies,
-      packageContent.optionalDependencies,
-      packageContent.peerDependencies,
-      packageContent.bundledDependencies,
-    ].some(hasKeys)) {
+    if (!haveAddedAnyDependencies()) {
       return null
     }
 
     return packageContent
   } catch (e) {
     if (paths.length > 0 && e.code === 'ENOENT') {
+      if (haveAddedAnyDependencies()) {
+        console.warn('[no-extraneous-dependencies] Some packageDir paths not exist.')
```

Comment:
what paths don't exist? can we tell them?

can we use `context.report` here, so it's a linting error?